elasticsearch/src/main/java/org/hibernate/search/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -378,14 +380,23 @@ public void performOperations(List<LuceneWork> workList, IndexingMonitor monitor
 
 	@Override
 	public void performStreamOperation(LuceneWork singleOperation, IndexingMonitor monitor, boolean forceAsync) {
-		if ( singleOperation == FlushLuceneWork.INSTANCE ) {
+		BackendRequest<?> request = singleOperation.acceptIndexWorkVisitor( visitor, monitor );
+		if ( singleOperation instanceof FlushLuceneWork ) {
 			requestProcessor.awaitAsyncProcessingCompletion();
+			executeBackendRequest( request, true );
 		}
 		else {
-			BackendRequest<?> request = singleOperation.acceptIndexWorkVisitor( visitor, monitor );
+			executeBackendRequest( request, false );
```

Comment:
Short story:

I left it ```false``` to maintain the behaviour currently in place: all work is async by default, so ```forceAsync``` is not used as before.

Long Story:

This is what the contract says about this flag:
```
@param forceAsync if true, the invocation will not block to wait for it being applied. 
When false this will depend on the backend configuration.
```

In the ```DirectotyBasedIndexManager``` there is a configuration ```worker.execution``` which assumes values ```{sync,async}``` that defines how works are applied to the index. In theory, the ```forceAsync``` flag should have no effect if the backend is already async. So assuming ```worker.execution=async``` and ```forceAsync=false```, what should happen? Having ```!forceAsync``` would make it sync... :disappointed: 

You may be asking why does it matter since the ```ElasticsearchIndexManager``` does not honor ```worker.execution```?  Should it honor it? I realized answering those question and implementing missing pieces were not directly related to this PR (which is about flush per entity) and maybe should be addressed in another JIRA, wdyt?